src/backend/utils/adt/ri_triggers.c

```diff
@@ -419,8 +418,8 @@ RI_FKey_check(TriggerData *trigdata)
 		appendStringInfoString(&querybuf, " FOR KEY SHARE OF x");
 
 		/* Prepare and save the plan */
-		RUN_AS_PSQL(qplan = ri_PlanCheck(querybuf.data, riinfo->nkeys, queryoids,
```

Comment:
though this is not ideal, let us not change existing code and apply checks at lower level, that might create unintended regressions in existing code